### DISCUSSION TITLE
LazyLoad components from an npm library

### DISCUSSION THREAD

**Discussion by iam-markus** [parcel-bundler/parcel#6050] [Q&A] [answered]
I want to lazy load most of my pages to reduce initial load size.
That works really well with Parcel + react, as long as the page is inside the project.
E.g. for the Registration I do:
`const Registration = lazy(() => import('./Registration'));`

I have now built a base library (npm lib) that is intended to share common functionality and common pages.

Would I be able to load a specific page / component in the same way from an npm bundle?
I want something like this:
const CommonPage = lazy(() => import('@mybundle/CommonPage'));

Do I maybe need to codesplit the CommonPage into an extra file?
If so, how do I do that? I already read the docs here: https://v2.parceljs.org/features/code-splitting/#using-dynamic-imports

This might be related: https://github.com/parcel-bundler/parcel/discussions/4661

**Comment by mischnic** [comment]
> const CommonPage = lazy(() => import('@mybundle/CommonPage'));

That should work already

**Reply by iam-markus** [reply]
Unfortunately I can't get it to work that way:

**Inside the npm library**

_./pages/CommonPage.js_
```
import React from 'react';

function CommonPage(props) {
    return <h1>I am pretty common</h1>
}

export default CommonPage;
```

_./index.js_
```
...
export * from './components/form/FormError';
export * from './components/form/Input';
export * from './components/form/Wizard';

export { default as CommonPage } from './pages/CommonPage';
```
(this one just exports all components + pages)
PS: it output one js file

**In my app**
I try to import CommonPage like this:
`const CommonPage = lazy(() => import('@org/mylib/CommonPage'));`

Which throws this error:
```
Build failed.
@parcel/core: Failed to resolve '@org/mylib/CommonPage' from './src/App'.
```

Is it maybe neccessary to build CommonPage as a separate bundle?

**Comment by iam-markus** [comment, marked as answer]
Alright, I figured it out on my own.
The key part is to export the component the right way.

**In the library:**
`export const CommonPage = React.lazy(() => import('./pages/CommonPage));`

**In the app:**
`import { CommonPage } from '@org/mylib';`
